### PR TITLE
Reverted BC change

### PR DESCRIPTION
**Pull request by RWOverdijk**
Revert collect method in interface, and gracefully skip if not implemented

### CODE REVIEW

File: src/AssetManager/Resolver/AggregateResolver.php [hunk 1]
```diff
@@ -60,6 +60,10 @@ public function collect()
         $collection = array();
 
         foreach ($this->queue as $resolver) {
+            if (!method_exists($resolver, 'collect') {
+                continue;
```
Comment (RWOverdijk) [inline src/AssetManager/Resolver/AggregateResolver.php]: Should I perhaps provide some sort of notice here?

Comment (Ocramius) [inline src/AssetManager/Resolver/AggregateResolver.php]: No, as it would be a BC break as well

**Comment by RWOverdijk** [comment]
I know the tests will fail because the mock is different now. I'm working on those currently.

**Comment by RWOverdijk** [comment]
@Ocramius CR please

File: tests/AssetManagerTest/Resolver/AggregateResolverTest.php [hunk 2]
```diff
@@ -8,6 +8,11 @@
 
 class AggregateResolverTest extends PHPUnit_Framework_TestCase
 {
+    public function setUp()
+    {
+        require_once __DIR__ . '/../../_files/ResolverWithCollect.php';
```
Comment (Ocramius) [inline tests/AssetManagerTest/Resolver/AggregateResolverTest.php]: Add it to `autoload-dev` in your `composer.json` instead.

Comment (Ocramius) [inline tests/AssetManagerTest/Resolver/AggregateResolverTest.php]: Actually, this class is not needed: just `$this->getMock('...ResolverInterface', ['collect'])`

File: tests/AssetManagerTest/Resolver/AggregateResolverTest.php [hunk 3]
```diff
@@ -69,5 +74,21 @@ public function testCollect()
         $this->assertContains('three', $collection);
 
         $this->assertCount(3, $collection);
+
+        /* Tests for interfaces that _don't_ implement the `collect` method. */
+        $resolver    = new AggregateResolver();
```
Comment (Ocramius) [inline tests/AssetManagerTest/Resolver/AggregateResolverTest.php]: Should be a separate test method.

File: tests/_files/ResolverWithCollect.php [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+use AssetManager\Resolver\ResolverInterface;
+
+class ResolverWithCollect implements ResolverInterface
```
Comment (Ocramius) [inline tests/_files/ResolverWithCollect.php]: Class not needed

File: tests/AssetManagerTest/Resolver/AggregateResolverTest.php (hunk 2 above)
Comment (RWOverdijk) [inline tests/AssetManagerTest/Resolver/AggregateResolverTest.php]: Do you want me to change [AssetManagerTest\Service\AggregateResolverServiceFactoryTest](https://github.com/RWOverdijk/AssetManager/blob/master/tests/AssetManagerTest/Service/AggregateResolverServiceFactoryTest.php#L18) as well?

Comment (RWOverdijk) [inline tests/AssetManagerTest/Resolver/AggregateResolverTest.php]: > Actually, this class is not needed: just $this->getMock('...ResolverInterface', ['collect'])
What do you mean? Does that add the method to the mock, even if it's not in the interface?

File: tests/AssetManagerTest/Resolver/AggregateResolverTest.php (hunk 3 above)
Comment (RWOverdijk) [inline tests/AssetManagerTest/Resolver/AggregateResolverTest.php]: The way it is now, each method has a test method. If I do this, I should do this all over the codebase.

File: tests/AssetManagerTest/Resolver/AggregateResolverTest.php (hunk 2 above)
Comment (Ocramius) [inline tests/AssetManagerTest/Resolver/AggregateResolverTest.php]: Correct

File: tests/AssetManagerTest/Resolver/AggregateResolverTest.php (hunk 3 above)
Comment (Ocramius) [inline tests/AssetManagerTest/Resolver/AggregateResolverTest.php]: This is a specific scenario and needs a specific test case with a specific name.

**Comment by RWOverdijk** [comment]
@Ocramius ready

**Comment by Ocramius** [comment]
:+1:

**Pull request merged**